client/dashboard/sites/settings-holiday-snow/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { siteBySlugQuery, siteSettingsMutation, siteSettingsQuery } from '@automattic/api-queries';
+import { useQuery, useSuspenseQuery, useMutation } from '@tanstack/react-query';
+import { notFound } from '@tanstack/react-router';
+import { __experimentalVStack as VStack, Button, CheckboxControl } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import Breadcrumbs from '../../app/breadcrumbs';
+import { ButtonStack } from '../../components/button-stack';
+import { Card, CardBody } from '../../components/card';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { isHolidaySnowAvailable } from './utils';
+import type { SiteSettings } from '@automattic/api-core';
+import type { Field, SimpleFormField } from '@wordpress/dataviews';
+
+const fields: Field< SiteSettings >[] = [
+	{
+		id: 'jetpack_holiday_snow_enabled',
+		label: __( 'Show falling snow on my site until January 4th' ),
+		Edit: ( { field, onChange, data, hideLabelFromVision } ) => {
+			const { id, getValue } = field;
+			return (
+				<CheckboxControl
+					__nextHasNoMarginBottom
+					label={ hideLabelFromVision ? '' : field.label }
+					checked={ getValue( { item: data } ) }
+					onChange={ () => {
+						onChange( { [ id ]: ! getValue( { item: data } ) } );
+					} }
+				/>
+			);
+		},
+	},
+];
+
+const form = {
+	layout: { type: 'regular' as const },
+	fields: [ { id: 'jetpack_holiday_snow_enabled' } as SimpleFormField ],
+};
+
+export default function HolidaySnowSettings( { siteSlug }: { siteSlug: string } ) {
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const { data: site } = useSuspenseQuery( siteBySlugQuery( siteSlug ) );
+	const { data } = useQuery( siteSettingsQuery( site.ID ) );
+	const mutation = useMutation( {
+		...siteSettingsMutation( site.ID ),
+		meta: {
+			snackbar: {
+				error: __( 'Failed to save holiday snow settings.' ),
+			},
+		},
+	} );
+
+	const [ formData, setFormData ] = useState( {
+		jetpack_holiday_snow_enabled: data?.jetpack_holiday_snow_enabled,
+	} );
+
+	if ( ! data ) {
+		return null;
+	}
+
+	if ( ! isHolidaySnowAvailable( site ) ) {
+		throw notFound();
+	}
+
+	const isDirty = Object.entries( formData ).some(
```

Comment:
`isDirty` isn't working very well for me. Sometimes `jetpack_holiday_snow_enabled` gets saved as `1` (checked) and `""` (unchecked) causing the `isDirty` comparison to mismatch. Refreshing the page fixes the issue. I've attached a recording.

https://github.com/user-attachments/assets/f18f09bf-c852-4aa1-a77c-fbf8e743315b

Reply:
Steps to reproduce:

1. Check the setting.
2. Refresh the page, "Save" button is enabled but the expectation is that it should be disabled.
3. Refresh the page, "Save" button is correctly disabled.

Same happens when unchecking the setting.